log/log.go

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0'
+ */
+
+package log
+
+import (
+	"fmt"
+	"log/syslog"
+	"os"
+)
+
+var slog *syslog.Writer
+
+// SetComponentName adds a component name to syslog tag as "my-snap.<component>"
+// The default tag is just "my-snap", read from the snap environment.
+// This function is NOT thread-safe. It should not be called concurrently with
+// the other logging functions of this package.
+func SetComponentName(component string) {
+	// update global value
+	tag = snapInstanceKey + "." + component
+	Debugf("Changing syslog tag to: %s", tag)
+
+	if err := setupSyslogWriter(tag); err != nil {
+		Errorf("Error changing syslog tag: %s", err)
+	}
+}
+
+// Debug writes the given input to syslog (sev=LOG_DEBUG) if snap `debug`
+// configuration option is set to `true`.
+// It formats similar to fmt.Sprint
+func Debug(a ...interface{}) {
+	if debug {
+		slog.Debug(fmt.Sprint(a...))
+	}
+}
+
+// Debugf writes the given input to syslog (sev=LOG_DEBUG) if snap `debug`
+// configuration option is set to `true`.
+// It formats similar to fmt.Sprintf
+func Debugf(format string, a ...interface{}) {
+	Debug(fmt.Sprintf(format, a...))
+}
+
+// Error writes the given input to syslog (sev=LOG_ERROR).
+// It formats similar to fmt.Sprint
+func Error(a ...interface{}) {
+	msg := fmt.Sprint(a...)
+	slog.Err(msg)
+	// print to stderr as well so that snap command prints them on non-zero exit
+	stderr(a...)
+}
+
+// Errorf writes the given input to syslog (sev=LOG_ERROR).
+// It formats similar to fmt.Sprintf
+func Errorf(format string, a ...interface{}) {
+	Error(fmt.Sprintf(format, a...))
+}
+
+// Info writes the given input to syslog (sev=LOG_INFO).
+// It formats similar to fmt.Sprint
+func Info(a ...interface{}) {
+	slog.Info(fmt.Sprint(a...))
+}
+
+// Infof writes the given input to syslog (sev=LOG_INFO).
+// It formats similar to fmt.Sprintf
+func Infof(format string, a ...interface{}) {
+	Info(fmt.Sprintf(format, a...))
+}
+
+// Warn writes the given input to syslog (sev=LOG_WARNING).
+// It formats similar to fmt.Sprint
+func Warn(a ...interface{}) {
+	slog.Err(fmt.Sprint(a...))
```

Comment:
`slog.Warning()`?

Reply:
I wanted to fix the #29 bug but somehow it got carried over.

Reply:
Good catch. It works fine now:
![image](https://user-images.githubusercontent.com/11150423/157720267-d5d8a38c-f2fb-4ac6-845b-0a5f41cbfa67.png)